src/app/helpers/message/messageDecrypt.ts

```diff
@@ -0,0 +1,61 @@
+import { decryptMIMEMessage, decryptMessageLegacy } from 'pmcrypto/lib/pmcrypto';
+import { MIME_TYPES } from 'proton-shared/lib/constants';
+import { c } from 'ttag';
+
+import { Message, MessageExtended } from '../../models/message';
+import { convert } from '../attachment/attachmentConverter';
+import { VERIFICATION_STATUS } from '../../constants';
+import { getDate } from './messages';
+import { AttachmentsCache } from '../../hooks/useAttachments';
+
+const getVerifiedStatus = (pmcryptoVerified: number, publicKeys: any) => {
+    const signedInvalid = VERIFICATION_STATUS.SIGNED_AND_INVALID;
+    const signedPubkey = VERIFICATION_STATUS.SIGNED_NO_PUB_KEY;
+    return !publicKeys.length && pmcryptoVerified === signedInvalid ? signedPubkey : pmcryptoVerified;
+};
+
+export const decryptMimeMessage = async (
+    message: Message,
+    privateKeys: any,
+    publicKeys: any,
+    attachmentsCache: AttachmentsCache
+): Promise<MessageExtended> => {
+    const headerFilename = c('Encrypted Headers').t`Encrypted Headers filename`;
+    const sender = (message.Sender || {}).Address;
+
+    const result = await decryptMIMEMessage({
+        message: message.Body,
+        messageDate: getDate(message),
+        privateKeys,
+        publicKeys,
+        headerFilename,
+        sender
+    });
+
+    const { body = c('Message empty').t`Message content if empty`, mimetype = MIME_TYPES.PLAINTEXT } =
+        (await result.getBody()) || {};
+
+    const verified = getVerifiedStatus(await result.verify(), publicKeys);
+
+    const attachments = convert(message, await result.getAttachments(), verified, attachmentsCache);
+    const encryptedSubject = await result.getEncryptedSubject();
+
+    return { raw: body, data: { ...message, Attachments: attachments }, verified, encryptedSubject, mimetype };
+};
+
+export const decryptLegacyMessage = async (
+    message: Message,
+    privateKeys: any,
+    publicKeys: any
+): Promise<MessageExtended> => {
+    const { data, verified: pmcryptoVerified = VERIFICATION_STATUS.NOT_SIGNED } = (await decryptMessageLegacy({
+        message: message.Body,
+        messageDate: getDate(message),
+        privateKeys,
+        publicKeys
+    })) as any;
```

Comment:
Any reason why we need `as any` here? If it's because `decryptMessageLegacy` does not define types, maybe we could define them there since it seems this problem does not exist for `decryptMIMEMessage`

Reply:
I had to try to remember... It comes from the type inference of TS over `pmcrypto`. TS compute this signature:
```typescript
function decryptMessageLegacy(options: any): Promise<{
    data: any;
    filename: any;
    verified: any;
    signatures: any;
} | {
    signature: number;
}>
```
So it refuses to consider `data` as a valid value to destructure. It's typically for this kind of things that I consider starting to write typings fro `pmcrypto`.